lib/datauri/datauri.go

```diff
@@ -134,11 +134,14 @@ func (u *HTTPURI) String() string {
 	return u.uri
 }
 
-func (u *HTTPURI) getRequest(ctx context.Context) (*http.Response, error) {
+func (u *HTTPURI) getRequest(ctx context.Context, extraHeaders ...headerModifier) (*http.Response, error) {
```

Comment:
A bit overkill - just want to allow callers passing header optionally.

Reply:
SGTM. This allows not only adding but mostly doing anything; eventually we can pas a map and simply add things to avoid one things to mess up with another (and possible the order in which things are applied).

This is a private method anyway, so not a big deal. SGTM.